concepts/0003-protocols/semver.md

```diff
@@ -1,93 +1,51 @@
-the message's minor version is earlier than their own preference. In such a case, the
-recipient should degrade gracefully to use the earlier version of the protocol. If the
-earlier version lacks important features, the recipient may optionally choose to send,
-in addition to a response, a `problem-report` with code `version-with-degraded-features`.
-
-If a recipient supports protocol X version 1.0, it should tentatively
-accept messages with later minor versions (e.g., 1.2). Message types that
-differ in only in minor version are guaranteed to be compatible *for the
-feature set of the earlier version*. That is, a 1.0-capable agent can support
-1.0 features using a 1.2 message, though of course it will lose any features
-that 1.2 added. Thus, accepting such a message could have two possible outcomes:
-
-1. The message at version 1.2 might look and behave exactly like it did at version
-1.0, in which case the message will process without any trouble.
+Within a given major version of a protocol, an agent should:
 
-2. The message might contain some fields that are unrecognized and need to be ignored.
+- specify a minimum minor version supported, defaulting to "0"
+- specify a current minor version that is fully supported and used by default
 
-In case 2, it is best practice for the recipient to send a `problem-report` that
-is a *warning*, not an *error*, announcing that some fields could not be processed
-(code = `fields-ignored-due-to-version-mismatch`). Such a message is *in addition
-to* any response that the protocol demands of the recipient.
+This leads to the following received message handling rules:
 
-If the recipient of a protocol's initial message generates a response, the response
-should use the latest major.minor protocol version that both parties support and
-know about. Generally, all messages after the first use only major.minor
+- message types received with a minor versions below the minimum may be answered with a `report-problem` with code `version-not-supported`
+- message types received with a minor version at or higher than the minimum supported and less than the current minor version are processed, ideally with a response using the same minor version of the received message
+  - The recipient may want to send a warning `report-problem` with code `version-with-degraded-features`
+- message types received with a minor version higher than the current minor version are processed with any unrecognized fields ignored
+  - The recipient may want to send a warning `report-problem` with code `fields-ignored-due-to-version-mismatch` 
 
-[![version negotiation matrix](version-negotiation-matrix.png)](
-https://docs.google.com/spreadsheets/d/1W5KYOqCCqmTeU4Z7XZQH9_6_0TeP5Vf5TtsOZmioyB0/edit#gid=0)
+As documented in the semvar documentation, these requirements may not apply when
+major version 0 is used. Specific rules associated each v0 protocol are left to
```

Comment:
That's not what semver says. Semver says that prior to v. 1.0, every minor version is a major change. I don't agree with the idea of letting each protocol handle this differently.

Reply:
Corrected.